app/scripts/modules/core/src/managed/ArtifactDetail.tsx

```diff
@@ -42,14 +48,97 @@ export const ArtifactDetail = ({
           {/* a short summary with actions/buttons will live here */}
           <div className="detail-section-right">{/* artifact metadata will live here */}</div>
         </div>
-        {environments.map(({ name }) => (
-          <div key={name}>
-            <h3>{name.toUpperCase()}</h3>
-            {resourcesByEnvironment[name].filter(shouldDisplayResource).map(resource => (
-              <ManagedResourceObject key={resource.id} resource={resource} />
-            ))}
-          </div>
-        ))}
+        {environments.map(({ name, state, deployedAt, replacedAt, replacedBy }) => {
+          const deployedAtMillis = DateTime.fromISO(deployedAt).toMillis();
+          const replacedAtMillis = DateTime.fromISO(replacedAt).toMillis();
+          const { version: replacedByPackageVersion, buildNumber: replacedByBuildNumber } =
+            parseName(replacedBy || '') || {};
+
+          return (
+            <div key={name}>
+              <h3>{name.toUpperCase()}</h3>
+              {state === 'deploying' && (
+                <NoticeCard
```

Comment:
i'll probably pull out all this card business into a component soon — needs to stabilize a little bit first.